### ISSUE TITLE
Outdated Documentation : WebSocketClient became WebSocketPeer

### ISSUE THREAD

**Issue by maelig-moreau** [godotengine/godot-docs#9138]
**Your Godot version:**
4.2 
**Issue description:**
The Page for deprecated WebSocketClient is still top research on Google and doesn't mention that WebSocketPeer is now in use.

**Fix**
- a clear indication that WebSocketPeer is the new standard.

- a stronger visual clue that the content is deprecated in 4.2 (the green text box on top is easy to miss since it's also used to indicate that the content is actually up to date, using orange would be better to warn the user)

**Comment by skyace65** [comment]
What page are you looking at? WebSocketClient was removed in 4.0 and is not part of the class reference.

**Comment by maelig-moreau** [comment]
Here :

https://docs.godotengine.org/en/stable/tutorials/networking/websocket.html

If you carefully read the top box you see that it hasn't been updated but I noticed that only after opening the issue (sorry!)

But I think it would be cool that the box wasn't green, I wouldn't have skipped reading it if it had a "error" type of colour see what I mean?

**Comment by ThuZHD** [comment]
Same problem for me, I want to get started with godot and websockets are perfect for me so please someone update the docs, thx

**Comment by maelig-moreau** [comment]
> Same problem for me, I want to get started with godot and websockets are perfect for me so please someone update the docs, thx

The page is here it's just not obvious and some like us stumble upon the old docs and don't notice it. 
Here you go : https://docs.godotengine.org/en/stable/classes/class_websocketpeer.html

**Comment by km19809** [comment]
I struggled to connect a Godot 4.2 client to the WebSocket echo server for 2 days, and I finally succeeded.
Here is the updated client code:
```gdscript
extends Node
# The URL we will connect to
@export var URL := "wss://echo.websocket.org"

# Our WebSocketPeer instance
var socket := WebSocketPeer.new()
var _socket_connected := false

func _ready():
	if socket.connect_to_url(URL) != OK:
		print("Unable to connect.")
		set_process(false)

func _process(_delta):
	# Call this in _process or _physics_process. 
	# Data transfer and state update will only happen when calling this function.
	socket.poll()
	var state = socket.get_ready_state()
	if state == WebSocketPeer.STATE_OPEN:
		if not _socket_connected:
			_socket_connected = true
			print("Connected with protocol: ", socket.get_selected_protocol())
			# Prefer `send_text` when sending a string over
			# `send` or `PacketPeer.put_packet`
			socket.send_text("Ping")
			 
		# Check if any packet is currently available in the ring buffer
		while socket.get_available_packet_count():
			# Print the received packet
			print("Got data from server: ", socket.get_packet().get_string_from_utf8())
	elif state == WebSocketPeer.STATE_CONNECTING:
		pass
	elif state == WebSocketPeer.STATE_CLOSING:
		pass
	elif state == WebSocketPeer.STATE_CLOSED:	
		# `get_close_code()` will return the received WebSocket close frame status code, 
		# or `-1` when the connection was not correctly notified
		# by the remote peer before closing the socket.
		# You should only call these methods below when `get_ready_state()` 
		# retruns WebSocketPeer.STATE_CLOSED
		var code = socket.get_close_code()
		var reason = socket.get_close_reason()
		print("WebSocket closed with code: %d, reason %s. Clean: %s" % [code, reason, code != -1])
		set_process(false)		

func _exit_tree():
	socket.close()

```

Result:
```
Connected with protocol: 
Got data from server: Request served by ...
Got data from server: Ping
```
(The second line is sent automatically by [websocket.org](wss://echo.websocket.org) )

However, because of changes in APIs, the result is not the same as the documentation.
For example, we will no longer specify protocols (ex. "lws-mirror-protocol").
Also, we don't use signals. 
Especially, there is no direct way to replace the `connection_error` and `connection_established` signals. 
So I used a flag variable to implement similar behavior.

I hope my experience will help to update the documentation.
